lib/downloader.rb

```diff
@@ -0,0 +1,35 @@
+# Defines how chromebrew downloads files.
+require_relative 'color'
+
+def notimplemented(package)
+  puts "#{package} has not been implemented as a downloader for Chromebrew yet. Defaulting to curl.".orange
+  @_crew_downloader = "curl -#LC -"
+  def download_file(url, filename)
+    system "#{@_crew_downloader} \'#{url}\' --output #{filename}"
+  end
+end
+
+case ENV["CREW_DOWNLOADER"]
+when 'aria2', 'aria2c'
+  def download_file(url, filename)
+    system "aria2c -c \'#{url}\' --out #{filename}"
+  end
+when 'curl', '', nil
+  def download_file(url, filename)
+    system "curl -#LC - \'#{url}\' --output #{filename}"
+  end
+when 'wget'
+  def download_file(url, filename)
```

Comment:
Make the filename argument optional? (extract filename from url if not given):
```ruby
def download_file ( url, filename = File.basename(url) )
```

Reply:
I did this a little differently, but it works :)